src/main/app/auth/user/oidc.ts

```diff
@@ -0,0 +1,46 @@
+import Axios from 'axios';
+import config from 'config';
+import jwt_decode from 'jwt-decode';
+import type { LoggerInstance } from 'winston';
+
+import { PageLink } from '../../../steps/urls';
+import { CaseApi } from '../../case/CaseApi';
+import { UserDetails } from '../../controller/AppRequest';
+
+export const CALLBACK_URL: PageLink = '/oauth2/callback';
+
+export const getRedirectUrl = (serviceUrl: string): string => {
+  const id: string = config.get('services.idam.clientID');
+  const loginUrl: string = config.get('services.idam.authorizationURL');
+  const callbackUrl = encodeURI(serviceUrl + CALLBACK_URL);
```

Comment:
Minor, but could be:

```suggestion
  const callbackUrl = encodeURI(`${serviceUrl}${CALLBACK_URL}`);
```